### PR TITLE
Rewrite of User Guide

### LINKED ISSUE #112: Check and refactor UG

### PR DESCRIPTION
**Pull request by itsme-zeix**
Rewrote User Guide to:
- Improve formatting consistency across UG
- Improve formatting using dropdowns/indentations
- Add undocumented commands
- Reduce length of commands details (caused by redundant/repeated information)

Images have yet to be included in the UG. Will do this in the next iteration once UI changes have been complete.

Closes #112.

### CODE REVIEW

**Review by FionaQY — APPROVED** [review]
LGTM. 

We will have to update this again later this week but this looks great! Good work!

**Review by iamdiluxedbutcooler — APPROVED** [review]
LGTM too! 
On my side will add-on some more details on `view` and `close` commands as per #105 above yours.

File: docs/UserGuide.md [hunk 1]
```diff
@@ -3,436 +3,713 @@ layout: page
 title: User Guide
 ---
 ![Banner](images/AgentAssistBanner.png) 
-## AgentAssist
-**Transform Your Credit Card Sales with AgentAssist** — the definitive desktop tool for bank agents. Merging the swift efficiency of a Command Line Interface (CLI) with the intuitive accessibility of a Graphical User Interface (GUI), AgentAssist propels your credit card sales into the fast lane. Tailored for the agile bank agent, this application lets you manage contact databases, track sales progress, and execute transactions with unprecedented speed. Maximize your productivity, minimize your response time, and amplify your sales performance. With AgentAssist, you're not just keeping up with the competitive world of credit card sales — _you're setting the pace_.
+# Welcome to the AgentAssist User Guide!
 
----
+The **AgentAssist User Guide** is here to help you unlock the full potential of **AgentAssist** and take your credit card sales to the next level. his guide offers clear, step-by-step instructions and practical examples to help you get the most out of the application.
```
Comment (ZShunRen) [inline docs/UserGuide.md:8]: Typo here should be `This` and not `his`.

File: docs/UserGuide.md [hunk 2]
```diff
@@ -3,436 +3,713 @@ layout: page
...
 
----
+Before you start using AgentAssist, there are a few prerequisites to ensure you get the most out of the application:
 
-## Using AgentAssist {#using-agentassist}
+### Familiarity with Keyboard Navigation
+AgentAssist is designed to enhance speed and efficiency, with a strong focus on **keyboard-based navigation**. While the application includes a Graphical User Interface (GUI), its full potential is unlocked when you use **keyboard commands**. Therefore, it is important to:
 
-AgentAssist's command box is your portal to managing contacts efficiently. Here’s how you can use it to streamline your tasks:
+- Be comfortable navigating applications using the keyboard.
```
Comment (ZShunRen) [inline docs/UserGuide.md:90]: Perhaps this line is not needed.

File: docs/UserGuide.md [hunk 3]
```diff
@@ -3,436 +3,713 @@ layout: page
...
+* **Contact Management**: 
+  * Manage your client details easily. Add, edit, and delete contacts to keep all your client information in one accessible place.
+* **Keyboard-only Navigation**: 
+  * Navigate through the application entirely via keyboard shortcuts, improving workflow efficiency.
+* **Multi-Level Filtering**: 
+  * Filter your data by multiple criteria to find exactly who you’re looking for.
+* **Auto-Save**: 
+  * Automatically saves your work as you go, ensuring data is updated without manual intervention.
+* **Effortless Data Import & Export**: 
+  * Import or export client and sales data in compatible formats for backups or use in other systems.
```
Comment (ZShunRen) [inline docs/UserGuide.md:77]: Nice labelling, but perhaps `Keyboard-only Navigation` is misleading and can be replaced with `Keyboard-Centric Navigation`, as they can still use the mouse for instance to close the app and to scroll.

File: docs/UserGuide.md [hunk 4]
```diff
@@ -3,436 +3,713 @@ layout: page
...
+
+Let's take a look at the structure in more detail:
 
-### Feature 2: Add New Customer {#add-new-customer}
+| **Components**  | **Description**                                                                                                                                               | **Example**                    |
+|:----------------|:--------------------------------------------------------------------------------------------------------------------------------------------------------------|:-------------------------------|
+| **Command**     | The action you want AgentAssist to perform.                                                                                                                   | `add`                          |
+| **Flag(s)**     | Modifiers that specify what kind of data is being handled. <br/><br/>Flag(s) are typically 1-2 letters followed by a backslash.                               | `n/`, `p/`, `r/`, `rn/`        |
+| **Argument(s)** | The values or inputs the command uses, such as client data or specific details. <br><br> The user guide may represent it as a placeholder using `<ARGUMENT>`. | `John Doe`, `john@example.com` |
```
Comment (ZShunRen) [inline docs/UserGuide.md:183]: Perhaps replace `The user guide` with `This guide`.

File: docs/UserGuide.md [hunk 5]
```diff
@@ -3,436 +3,713 @@ layout: page
...
+| `delete`    | Removes a client from the system.                      |
+| `list`      | Displays all clients currently stored in the system.   |
+| `filter`    | Filters clients based on specified criteria            |
+| `clear`     | Deletes all clients from the system.                   |
+| `help`      | Displays a list of available commands and their usage. |
+| `exit`      | Exits the AgentAssist application.                     |
 
-**Handling Duplicates:**  
-If a customer with the same name, email, job, and income is already saved, you'll get a message: "This customer is already saved as a contact."
+Refer to the [Commands Section](#commands-section) for more comprehensive details of each command.
```
Comment (ZShunRen) [inline docs/UserGuide.md:210]: Nice link here.

File: docs/UserGuide.md [hunk 6]
```diff
@@ -3,436 +3,713 @@ layout: page
...
+| `j/`     | `job`            |
+| `i/`     | `income`         |
+| `t/`     | `tier`           |
+| `r/`     | `remark`         |
+| `ra/`    | `remark append`  |
+| `rn/`    | `remark new`     |
+
+> 💡 **Pro Tip:**
+> 
+> Flags are typically derived from the first letter of their corresponding data type (e.g., `n/` for `name`), making them easy to remember!
```
Comment (ZShunRen) [inline docs/UserGuide.md:234]: Nice!

File: docs/UserGuide.md [hunk 7]
```diff
@@ -3,436 +3,713 @@ layout: page
...
+| `i/`     | `<INCOME>`            | The client's annual income                     | Positive number or zero <br/> • Cannot include commas and decimal points<br/> • Must be numeric | ❌                     |
+| `t/`     | `<TIER>`              | The client's assigned tier level               | Predefined tiers:<br/> • Gold, Silver, Bronze, Reject                                           | ✔️                    |
+| `r/`     | `<REMARK>`            | General remarks about the client               | No specific restrictions                                                                        | ❌                     |
+| `ra/`    | `<REMARK TO APPEND>`  | Append information to the existing remark      | No specific restrictions                                                                        | ❌                     |
+| `rn/`    | `<NEW REMARK>`        | Replaces the existing remark with a new remark | No specific restrictions                                                                        | ❌                     |
```
Comment (ZShunRen) [inline docs/UserGuide.md:253]: Perhaps in line with what pleng suggested, don't state that there are `no specific restrictions` rather just state `any combination of characters` but state what they can put first before stating what they cannot put.

File: docs/UserGuide.md [hunk 8]
```diff
@@ -3,436 +3,713 @@ layout: page
...
+| `r/`     | `<REMARK>`            | General remarks about the client               | No specific restrictions                                                                        | ❌                     |
+| `ra/`    | `<REMARK TO APPEND>`  | Append information to the existing remark      | No specific restrictions                                                                        | ❌                     |
+| `rn/`    | `<NEW REMARK>`        | Replaces the existing remark with a new remark | No specific restrictions                                                                        | ❌                     |    
 
-### Feature 3: Remove Old Customer {#remove-old-customer}
+> 💡 **Pro Tip:**
+> 
+> Ensure every flag is followed by a valid argument!
+>
+> Providing a flag without an accompanying argument will result in an error and prevent the command from executing properly.
```
Comment (ZShunRen) [inline docs/UserGuide.md:259]: Nice, but perhaps replcae `every flag` with `every specified flag`.

File: docs/UserGuide.md [hunk 9]
```diff
@@ -3,436 +3,713 @@ layout: page
...
+
+**Editing a Client's Information**
+* `edit 1 p/12345678`: Updates the phone number of the first client in your list to `12345678`.
+* `edit 4 rn/Updated remarks here`: Replaces the remarks of the fourth client with "Updated remarks here".
+
+**Removing a Client**
+* `delete 3`: Removes the third client from your list. Ensure you have the correct index to avoid deleting the wrong client. 
+
+**Searching for a Client**
+* `filter n/Jane`: Finds all client named Jane in your database. It’s a powerful tool for quickly locating clients or filtering for a specific type of client.
```
Comment (ZShunRen) [inline docs/UserGuide.md:279]: Should be `Finds all clients named Jane`.

File: docs/UserGuide.md [hunk 10]
```diff
@@ -3,436 +3,713 @@ layout: page
...
+
+### Command Syntax
+
+When reading commands, there are certain syntax conventions that help indicate how to use them:
+
+- **`< >` (Angle Brackets):**  
+  Text enclosed in angle brackets represents a **placeholder** for the actual value you need to provide. For example, `<NAME>` should be replaced by the client's actual name, such as "John Doe."
+
+- **`[ ]` (Square Brackets):**  
+  Components enclosed in square brackets are **optional**. You can choose to include them if necessary, but they are not required for the command to execute. For instance, `[t/ <TIER>]` means that the credit card tier is optional, and if omitted, a default value will be used.
```
Comment (ZShunRen) [inline docs/UserGuide.md:310]: Nice explanation here for non-technical users.

File: docs/UserGuide.md [hunk 11]
```diff
@@ -3,436 +3,713 @@ layout: page
...
+    ```
+    Please verify that your input is in the correct format. Include the following details: n/ <NAME> p/ <PHONE> e/ <EMAIL> a/ <ADDRESS> j/ <JOBNAME> i/ <INCOME> [t/ <TIER>] [rn/ <REMARK>].
+    ```
 
-| Parameter | Expected Format               | Explanation                                                                                                                                 |
-|-----------|-------------------------------|---------------------------------------------------------------------------------------------------------------------------------------------|
-| INDEX     | Integer (1 to the last INDEX) | The INDEX must be a valid integer within the registered range (either the original list or any filtered list after using `filter` command). |
+> **Note on Duplicates:**
+>
+> AgentAssist will prevent duplicate entries if a client with the **same name, email, job, and income** is already saved.
```
Comment (ZShunRen) [inline docs/UserGuide.md:362]: This has been amended, refer to #108. But equality is now stricter, with same name, email and phone being the conditions for equality.

File: docs/UserGuide.md [hunk 12]
```diff
@@ -3,436 +3,713 @@ layout: page
...
+| **Delete Existing Client** | `delete <INDEX>`                                                                                                             | `delete 69`                                                                                               |
+| **Edit Existing Client**   | `edit <INDEX> n/<NAME> p/<PHONE> e/<EMAIL> a/<ADDRESS> j/<JOB> i/<INCOME> [t/<TIER>] [rn/<NEW REMARK>] [ra/<APPEND REMARK>]` | `edit 69 n/ GORDON MOORE p/ 77337733 e/ gmoore_new@ntu.sg a/ COM3 j/ doctor i/ 1000000000 ra/ added info` |
+| **List All Clients**       | `list`                                                                                                                       | `list`                                                                                                    |
+| **Filter Client List**     | `filter n/<NAME> p/<PHONE> e/<EMAIL> a/<ADDRESS> j/<JOB> r/<REMARK> t/<TIER> i/<INCOME>`                                     | `filter n/ GORDON MOORE j/ doctor t/ gold`                                                                |
```
Comment (ZShunRen) [inline docs/UserGuide.md:707]: Perhaps you can use the optional parameters format for the flags in `filter` command.

**Review by ZShunRen — APPROVED** [review]
Overall LGTM, just a few grammatical mistakes and possible improvements. Great work!

File: docs/UserGuide.md (hunk 2 above)
Comment (itsme-zeix) [reply docs/UserGuide.md:90]: Good catch!

File: docs/UserGuide.md (hunk 1 above)
Comment (itsme-zeix) [reply docs/UserGuide.md:8]: Nice catch!

File: docs/UserGuide.md (hunk 2 above)
Comment (itsme-zeix) [reply docs/UserGuide.md:90]: Agreed, the title of that section implies it anyways.

File: docs/UserGuide.md (hunk 3 above)
Comment (itsme-zeix) [reply docs/UserGuide.md:77]: Yup, I agree with this, `keyboard-centric` does fit this scenario better.

File: docs/UserGuide.md (hunk 4 above)
Comment (itsme-zeix) [reply docs/UserGuide.md:183]: Included this change for clarity.

File: docs/UserGuide.md (hunk 8 above)
Comment (itsme-zeix) [reply docs/UserGuide.md:259]: I feel that `every flag` is concise and to the point - adding "specified" might create unnecessary doubt or confusion about whether there are some flags that don't need arguments.

File: docs/UserGuide.md (hunk 9 above)
Comment (itsme-zeix) [reply docs/UserGuide.md:279]: Nice catch! Fixed.

File: docs/UserGuide.md (hunk 11 above)
Comment (itsme-zeix) [reply docs/UserGuide.md:362]: Appreciate the catch, amended!

File: docs/UserGuide.md (hunk 12 above)
Comment (itsme-zeix) [reply docs/UserGuide.md:707]: Great idea, I've included this change.

**Pull request merged**